File: app/src/main/java/com/github/panda_io/micro_panda/ast/expression/Parentheses.java
package com.github.panda_io.micro_panda.ast.expression;

import com.github.panda_io.micro_panda.ast.type.Type;
import com.github.panda_io.micro_panda.ast.Context;

public class Parentheses extends Expression {
    public Expression expression;
    
	public boolean isLvalue() {
		return this.expression.isLvalue();
	}

    public void validate(Context context, Type expected) {
        this.expression.validate(context, expected);
        this.constant = this.expression.isConstant();
        this.type = this.expression.type;
    }
}

File: MicroPanda/AST/Expression/Parentheses.cs
namespace MicroPanda.AST.Expression;

using Type;

internal class Parentheses : Expression
{
    internal Expression? Expression { get; set; }

    internal override void Validate(Context context, Type? expected)
    {
        Expression!.Validate(context, expected);
        _const = Expression.Const;
        _type = Expression.Type;
    }
}